CollateralAccount/CollateralAccount.md

```diff
@@ -0,0 +1,106 @@
+1. If refactoring can unblock the wallet, prompt the user to send a transaction to refactor their UTXO (and pay the tx fee associated with it) before sending their transaction. If refactoring can't unblock the wallet, tell the user they will need more ADA to call this smart contract
+
+This however, causes the following problems:
+1. This may cause users to risk more collateral than they are comfortable with. In general, software should run smart contracts locally to detect if a transaction would fail before sending and alert the user to avoid consuming the collateral, but relying on this is not ideal.
+1. Although the refactor transaction can be created under the hood for in-software wallet, it will confused users when it shows up on their transaction history. For hardware wallets, it will require explicit approval from the user which is also possibly confusing.
+1. Altough wallets can try and always pick UTXOs to make sure there is always a valid UTXO entry for collateral, it can't be guaranteed because the wallet state can always be changed by any dApp or other wallet that doesn't use the same logic to guarantee the presence of satisfactory collateral input
+1. Multisig wallets don't have any non-script addresses in general and so would need some separate solution
+
+So with this, we can see the naive solution is complicated while still having issues and an unintuitive user experience.
+
+
+
+# Specification
+
+Recall that [CIP1852](../CIP-1852) specifies the following derivation path
+
+```
+m / purpose' / coin_type' / account' / chain / address_index
+```
+
+We set `chain=3` to indicate the *collateral chain*.
+
+## *address_index* value
+
+Wallets MUST only use `address_index=0` for this specification. Since a single address can contain multiple UTXO entries, there is no need to derive other addresses (using more addresses would not provide any privacy benefit and would complicate address discovery).
+
+We will call this specific derivation the *collateral account*.
+
+## New solution for managing collateral
+
+Wallets SHOULD only use UTXO entries of the collateral account as collateral in the transaction. If the collateral is insufficient, the wallet should tell the user to send more ADA to their collateral account.
+
+The benefits of this solution are as follows:
+1. Transactions in the history to add collateral can be clearly marked as such by wallet software.
+1. Prompting the user to add ADA to their collateral account doesn't require explaining users about how UTXO works under the hood compared to the refactoring option.
+1. No change required to input selection algorithms
+1. Multisig (or other script-based wallets) are handled the same way as regular wallets
+1. User only risks as much collateral as their are comfortable with
+
+However, this solution also comes with downsides:
```

Comment:
If we decide we don't want to explicitly tell the users about the collateral address existence, that's fine. It can be like how we have external & internal addresses without the user really knowing about their existence.

> but is fortunately rare
I don't think this is that rare. When I was working at EMURGO we usually had a few support tickets about this topic per day due to NFTs and the presence of tokens will only increase going forward -- especially with Alonzo